### DISCUSSION TITLE
ScriptReference/MeshRenderer-additionalVertexStreams

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#230] [Notes]
# ScriptReference/MeshRenderer-additionalVertexStreams

https://docs.unity3d.com/ScriptReference/MeshRenderer-additionalVertexStreams.html

**Comment by mvi** [comment]
Because additionalVertexStreams takes a Mesh you may find if you want to set a single channel (such as Colors) that you get an error about there being no vertices (positions) set. In order to side step this issue set [`SetVertexBufferParams`](https://docs.unity3d.com/ScriptReference/Mesh.SetVertexBufferParams.html) with an attribute layout that only has the attributes you need.
